kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/ApiVersionsDowngradeIT.java

```diff
@@ -133,6 +140,50 @@ private void doProxyRestrictedToOlderApiVersion(KafkaCluster cluster, Map<String
         }
     }
 
+    /**
+     * In this test, we verify the assumption that the Kafka Client is capable
+     * of negotiating down api-versions version twice.  This test uses an
+     * old version of Redpanda (that supports max v2) and the Proxy (restricted
+     * to v3).  The Kafka Client is using v4.
+     * <br>
+     * Client will first make a v4 request.  The proxy's response will cause the client
+     * to try again at v3.  The broker will then cause the client to try a third time at v2.
+     * This request will satisfy all parties and the connection establishment will continue.
+     * All this occurs on a single connection.
+     */
+    @Test
+    @EnabledIf(value = "isDockerAvailable", disabledReason = "docker unavailable")
+    void clientAheadOfProxyWhichIsAheadOfBroker() {
+
+        try (var redpanda = createRedpanda(OLD_REDPANDA_USING_API_VER0_2)) {
```

Comment:
It'd be preferable to test with Kafka 2.4.   We haven't got a kafka-native 2.4 image right now so I used redpanda instead.  I'll see if I can make time to do a kafka native 2.4 but I don't want to block this PR from moving forward.